_data-prepper/pipelines/configuration/sources/documentdb.md

```diff
@@ -0,0 +1,94 @@
+`s3_prefix` | No | String  | An optional key prefix in Amazon S3. By default, there is no key prefix.
+`s3_region` | No | String  | The AWS region where the S3 bucket resides.
+`aws` | Yes | AWS | The AWS configuration. See [aws](#aws) for more information.
+`id_key` | No | String  | When specified, the `_id` field from DocumentDB will be set to the key name specified by `id_key`. You can use this when need more information beyond an ObjectId string saved to your sink. By default, the `_id` is not made as part of the Event.
+`direct_connection` | No | Boolean  | When `true`, the MongoDB driver will connect directly to the specified DocumentDB server or servers without discovering and connecting to the entire replica set. Defaults to `true`.
+`read_preference` | No | String  | Determines how to read from DocumentDB. See [Read Preference Modes](https://www.mongodb.com/docs/v3.6/reference/read-preference/#read-preference-modes) for details. Defaults to `primaryPreferred`.
+`disable_s3_read_for_leader` | No | Boolean  | When `true`, the current leader node will not read from S3. It will only do the work of reading the stream. Defaults to `false`.
+`partition_acknowledgment_timeout` | No | Duration  | Configures how long a node will hold onto a partition. Defaults to `2h`.
+`acknowledgments` | No | Boolean  | When `true`, enables [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines#end-to-end-acknowledgments) on this source after events are sent to the sinks.
+`insecure` | No | Boolean | Disables TLS. Defaults to `false`. Do not use this value in production.
+`ssl_insecure_disable_verification` | No | Boolean | Disables TLS hostname verification. Defaults to `false`. Do not use this value in production. Use the `trust_store_file_path` to verify the hostname.
+
+### authentication
+
+The following parameters allow you to configure authentication to the DocumentDB cluster.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`username` | Yes | String | The username to use when authenticating with the DocumentDB cluster. Supports automatic refresh.
+`password` | Yes | String | The password to use when authenticating with the DocumentDB cluster. Supports automatic refresh.
+
+
+### collection
+
+The following parameters allow you to configure the collection to read from the DocumentDB cluster.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`collection` | Yes | String | The name of the collection
+`export` | No | Boolean | Whether to include an export or full load. Defaults to `true`.
+`stream` | No | Boolean | Whether to enable a stream. Defaults to `true`.
+`partition_count` | No | Integer | Defines the number of partitions to create in Amazon S3. Defaults to `100`.
+`export_batch_size` | No | Integer | Defaults to `10,000`.
+`stream_batch_size` | No | Integer | Defaults to `1,000`.
+
+## aws
+
```

Comment:
The following parameters enable you to <what do they enable user to do?>

Reply:
@vagimeli , Please see my latest revision where I answered this.